cluster/v1alpha1/helpers.go

```diff
@@ -0,0 +1,379 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/clock"
+	clusterv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+)
+
+var RolloutClock = clock.Clock(clock.RealClock{})
+var maxTimeDuration = time.Duration(math.MaxInt64)
+
+// RolloutStatus represents the status of a rollout operation.
+type RolloutStatus int
+
+const (
+	// ToApply indicates that the resource's desired status has not been applied yet.
+	ToApply RolloutStatus = iota
+	// Progressing indicates that the resource's desired status is applied and last applied status is not updated.
+	Progressing
+	// Succeed indicates that the resource's desired status is applied and last applied status is successful.
+	Succeed
```

Comment:
Nit: I think this would fit better as "Succeeded"

Reply:
I've filed a PR in case there's an interest in updating this: https://github.com/open-cluster-management-io/api/pull/267